src/fastoad/io/tests/test_data_file.py

```diff
@@ -46,18 +46,39 @@ def write_variables(self, data_source: Union[str, IO], variables: VariableList):
         self.variables.update(variables, add_variables=True)
 
 
-def test(cleanup):
+def test_DataFile(cleanup):
+    variables_ref = VariableList([Variable("data:foo", value=5), Variable("data:bar", value=10)])
+
     file_path = pth.join(RESULTS_FOLDER_PATH, "dummy_data_file.xml")
-    variables_1 = DataFile(file_path)
-    assert len(variables_1) == 0
+    with pytest.raises(FileNotFoundError) as exc_info:
+        _ = DataFile(file_path)
+    assert exc_info.value.args[0] == f'File "{file_path}" is unavailable for reading.'
```

Comment:
I suggest adding a comment that explains what all the assertion do. On this one, which I assume raises an error when reading a non existant file could be 
```python
# Tests that an error is raised when opening non_existent file (Issue #429)
```

Reply:
You understood the purpose, which tends to show that the comment is not mandatory :)

Generally speaking, comments that just paraphrase the code are considered bad practice.
Comments should be used only to explain why a given solution has been chose, or to help understanding a particularly complex piece of code (which is generally the sign that the code has problems, but as mere humans, we sometimes have to cope with that).

Here, the only additional information I see would be the source bug. Though I am willing to mention the issue for very peculiar bugs, the checked feature here is very basic, so I do not see the need.

Do you agree?

Reply:
Indeed, but because I read the issue beforehand.
Another question out of curiosity, is it assumed that basic users (as opposed to devs) won't go into the tests for informations on what is possible to do (meaning we can assume that the people who will read this have a better understanding of what is done) ? 
I wasn't sure that the good practice also applied to tests, but now I know :)
We can resolve this then.